NitroxInstallerActions/Patches/FirewallRules.cs

```diff
@@ -46,112 +27,146 @@ string RunNetshCommand(string Argument)
+        public string name;
+        public bool isInstalled;
+        string[] GetAddRemoveCommand(string RuleName, string GamePath, int Port = 11000)
         {
-            bool status = true;
-            foreach(string item in RulesArray)
+            string[] commands = new string[2] { "", "" };
+            if (RuleName == "PortsUDP")
             {
-                string command = "nothing";
-                if (item == "PortsUDP")
-                {
-                    command= "advfirewall firewall add rule name=\x22Nitrox Multiplayer Mod UDP\x22 dir=in action=allow protocol=UDP localport=" + PortNumber;
-                }
-                if (item == "ProgramServer")
-                {
-                    command= "advfirewall firewall add rule name=\x22Nitrox Multiplayer Mod Server\x22 dir=in action=allow program=\x22" + Path.Combine(SubPath, "SubServer", "NitroxServer.exe") + "\x22 enable=yes";
-                }
-                if (item == "ProgramClient")
-                {
-                    command= "advfirewall firewall add rule name=\x22Nitrox Multiplayer Mod Client\x22 dir=in action=allow program=\x22" + Path.Combine(SubPath, "Subnautica.exe") + "\x22 enable=yes";
-                }
-                if (!(command == "nothing"))
-                {
-                    string output = RunNetshCommand(command);
-                    if (!output.Contains("Ok."))
-                    {
-                        status = false;
-                    }
-                }
+                commands[0] = "advfirewall firewall add rule name=\x22Nitrox Multiplayer Mod UDP\x22 dir=in action=allow protocol=UDP localport=" + Port;
+                commands[1] = "advfirewall firewall delete rule name=\x22Nitrox Multiplayer Mod UDP\x22 protocol=UDP localport=" + Port;
             }
-            return status;
+            if (RuleName == "ProgramServer")
+            {
+                commands[0] = "advfirewall firewall add rule name=\x22Nitrox Multiplayer Mod Server\x22 dir=in action=allow program=\x22" + Path.Combine(GamePath, "SubServer", "NitroxServer.exe") + "\x22 enable=yes";
+                commands[1] = "advfirewall firewall delete rule name=\x22Nitrox Multiplayer Mod Server\x22";
+            }
+            if (RuleName == "ProgramClient")
+            {
+                commands[0] = "advfirewall firewall add rule name=\x22Nitrox Multiplayer Mod Client\x22 dir=in action=allow program=\x22" + Path.Combine(GamePath, "Subnautica.exe") + "\x22 enable=yes";
+                commands[1] = "advfirewall firewall delete rule name=\x22Nitrox Multiplayer Mod Client\x22";
+            }
+            return commands;
         }
-        bool RemoveFirewallRules(List<string> RulesArray, int PortNumber)
+        bool GetRuleInstalled(string RuleName)
         {
-            bool status=true;
-            foreach(string item in RulesArray)
+            string command = "", output = "";
+            if (RuleName == "PortsUDP")
```

Comment:
Use a switch or else if statements. There's only one if-statement that gets triggered.